ui/components/component-library/avatar-icon/avatar-icon.types.ts

```diff
@@ -0,0 +1,35 @@
+import { IconColor, TextColor } from '../../../helpers/constants/design-system';
+import { IconName, IconProps } from '../icon';
+import { AvatarBaseProps } from '../avatar-base/avatar-base.types';
+
+export enum AvatarIconSize {
+  Xs = 'xs',
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+  Xl = 'xl',
+}
+
+export interface AvatarIconProps extends Omit<AvatarBaseProps, 'color'> {
+  /**
+   * The name of the icon to display. Should be one of IconName
+   */
+  iconName: IconName;
+  /**
+   * Props for the icon inside AvatarIcon. All Icon props can be used
+   */
+  iconProps?: Omit<IconProps, 'name'> & {
+    'data-testid'?: string;
+  };
+  /**
+   * The size of the AvatarIcon
+   * Possible values could be 'AvatarIconSize.Xs' 16px, 'AvatarIconSize.Sm' 24px, 'AvatarIconSize.Md' 32px, 'AvatarIconSize.Lg' 40px, 'AvatarIconSize.Xl' 48px
+   * Defaults to AvatarIconSize.Md
+   */
+  size?: AvatarIconSize;
+  /**
+   * The color of the text inside the AvatarIcon
+   * Defaults to TextColor.primaryDefault
+   */
+  color?: TextColor | IconColor;
```

Comment:
Hi @georgewrmarshall, Should I add union of TextColor and IconColor to 'color' of AvatarBaseProps ? Or should I just remove all instances of IconColor?

Reply:
Hmm this is because for most Avatars when we think of color we are thinking of the color of the text but here it is for the icon. I think it would be ok to add the union of TextColor and IconColor to `color` of AvatarBaseProps